commons/zenoh-codec/src/transport/init.rs

```diff
@@ -98,6 +100,10 @@ where
             n_exts -= 1;
             self.write(&mut *writer, (qos, n_exts != 0))?;
         }
+        if let Some(qos_optimized) = ext_qos_optimized.as_ref() {
```

Comment:
By doing so, both `ext_qos` and `ext_qos_optimized` could be sent on the network.
If `ext_qos_optimized` then I don't see much the need of sending as well `ext_qos`.

Reply:
Not quite:

https://github.com/eclipse-zenoh/zenoh/blob/a6d3796457d296acdb15957a938fd29c2ee00473/io/zenoh-transport/src/unicast/establishment/ext/qos.rs#L155-L171

https://github.com/eclipse-zenoh/zenoh/blob/a6d3796457d296acdb15957a938fd29c2ee00473/io/zenoh-transport/src/unicast/establishment/ext/qos.rs#L173-L188

The invariant is maintained in the `qos` module, doing more checks here would be redundant. I don't think it's the encoder's concern.

Reply:
That's fair.